framework/include/executors/Executor.h

```diff
@@ -0,0 +1,166 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Executioner.h"
+#include "ExecutorInterface.h"
+
+#include <string>
+
+class Problem;
+class Executor;
+
+/// The Executor class directs the execution flow of simulations.  It manages
+/// outputting, time stepping, mesh adaptivity, solve sequencing, multiapp
+/// execution, etc.  Users can compose executor objects in an input file using
+/// the same pattern as with mesh generators.  Subclass here and implement your
+/// own run function if you need special or fancy functionality that
+/// isn't supported by the default executors available.
+class Executor : public Executioner, public ExecutorInterface
```

Comment:
```suggestion
class Executor : public Executioner, public ExecutorInterface
```
can we avoid inheriting the Executioner?
Stuff like calls to executor.feProblem should really not be possible

Reply:
well it makes sense for this particular one, since it has to do the solve. Does it make sense for a Refine one?

Reply:
I believe that this is just a transient situation.  It was the easiest way to "rig" Executors into the current system.

Reply:
Yep.  They could be separated.  Since eventually executioners will be killed, it didn't seem worth the trouble for now.